xla/service/gpu/tests/gemm_rewrite_test.cc

```diff
@@ -5431,6 +5592,230 @@ TEST_P(ParameterizedFp8GemmRewriteTest, ScaledABAlphaDF8) {
       )");
 }
 
+TEST_P(ParameterizedFp8GemmRewriteTest, ScaledABUnscaledDReluWithAuxF8) {
+#if GOOGLE_CUDA && CUDA_VERSION < 12000
+  GTEST_SKIP() << "F8 gemm rewrite is only supported in CUDA 12 and above.";
+#endif  // CUDA_VERSION < 12000
+
+#if TENSORFLOW_USE_ROCM && TF_ROCM_VERSION < 60000
+  GTEST_SKIP() << "F8 gemm rewrite is only supported in ROCm 6.0 and above.";
+#endif  // TF_ROCM_VERSION < 60000
+
+if (CudaOrRocmCheck(Switch::False, Switch::True)) {
+    GTEST_SKIP() << "Unsupported blas-lt epilogue on ROCM";
+}
+
+  const char* hlo_text = R"(
+    HloModule test
+
+    ENTRY test {
+      x = <<F8E4M3>>[128,128] parameter(0)
+      y = <<F8E4M3>>[128,128] parameter(1)
+      x_f32 = f32[128,128] convert(x)
+      y_f32 = f32[128,128] convert(y)
+      x_scale = f32[] parameter(2)
+      y_scale = f32[] parameter(3)
+      x_scale_bcast = f32[128,128] broadcast(x_scale), dimensions={}
+      y_scale_bcast = f32[128,128] broadcast(y_scale), dimensions={}
+      x_unscaled = f32[128,128] multiply(x_f32, x_scale_bcast)
+      y_unscaled = f32[128,128] multiply(y_f32, y_scale_bcast)
+      const.0 = f32[] constant(0)
+      bcast.0 = f32[128,128] broadcast(const.0), dimensions={}
+      dot = f32[128,128] dot(x_unscaled, y_unscaled), lhs_contracting_dims={1}, rhs_contracting_dims={0}
+      c = f32[128,128] parameter(8)
+      maximum.1 = f32[128,128] maximum(dot, bcast.0)
+      dot.1 = f32[128,128] dot(maximum.1, c), lhs_contracting_dims={1}, rhs_contracting_dims={0}
```

Comment:
I don't think dot.1 has any purpose here. I previously suggested testing this by having a path from the forward dot to the backwards dot, but this doesn't accomplish that. Instead, you can replaces uses of `a` with `dot` to accomplish that.